### PR TITLE
search implemented in places

### PR DESCRIPTION
**Pull request by smtrz**
I have implemented search functionality in the places.. please review.

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -30,6 +30,7 @@ android {
         sourceCompatibility JavaVersion.VERSION_1_8
         targetCompatibility JavaVersion.VERSION_1_8
     }
+
```
Comment (Swati4star) [inline app/build.gradle]: remove this file from commit

File: app/src/main/java/io/github/project_travel_mate/destinations/description/PlacesOnMapActivity.java [hunk 2]
```diff
@@ -371,13 +384,31 @@ public boolean onMarkerClick(Marker marker, MapView mapView) {
...
+
+    @Override
+    public void onTextChanged(CharSequence s, int start, int before, int count) {
+
+    }
+
+    @Override
+    public void afterTextChanged(Editable editable) {
+
+        //  if(editable.toString().length()) {
```
Comment (Swati4star) [inline app/src/main/java/io/github/project_travel_mate/destinations/description/PlacesOnMapActivity.java]: remove commented string

File: app/src/main/java/io/github/project_travel_mate/destinations/description/PlacesOnMapActivity.java [hunk 3]
```diff
@@ -501,4 +537,28 @@ public ViewHolder(View view) {
 
         }
     }
+
+    private void filter(String text) {
```
Comment (Swati4star) [inline app/src/main/java/io/github/project_travel_mate/destinations/description/PlacesOnMapActivity.java]: add javadoc style comments

File: app/src/main/java/io/github/project_travel_mate/destinations/description/PlacesOnMapActivity.java [hunk 4]
```diff
@@ -501,4 +537,28 @@ public ViewHolder(View view) {
...
+        //new array list that will hold the filtered data
```
Comment (Swati4star) [inline app/src/main/java/io/github/project_travel_mate/destinations/description/PlacesOnMapActivity.java]: remove this and next line comment

File: app/src/main/java/io/github/project_travel_mate/destinations/description/PlacesOnMapActivity.java (hunk 3 above)
Comment (Swati4star) [inline app/src/main/java/io/github/project_travel_mate/destinations/description/PlacesOnMapActivity.java]: change `text` to `searchtxt`

File: app/src/main/java/io/github/project_travel_mate/destinations/description/PlacesOnMapActivity.java [hunk 5]
```diff
@@ -501,4 +537,28 @@ public ViewHolder(View view) {
...
+        //
+        final JSONArray mFeedItems_filtered = new JSONArray();
```
Comment (Swati4star) [inline app/src/main/java/io/github/project_travel_mate/destinations/description/PlacesOnMapActivity.java]: Don't use m prefix here

`filteredFeedItems`

Comment (smtrz) [reply app/src/main/java/io/github/project_travel_mate/destinations/description/PlacesOnMapActivity.java]: done.

**Comment by Swati4star** [comment]
Also, please share a screenshot with the changes

File: app/build.gradle (hunk 1 above)
Comment (Swati4star) [reply app/build.gradle]: not resolved yet

File: app/src/main/java/io/github/project_travel_mate/destinations/description/PlacesOnMapActivity.java [hunk 6]
```diff
@@ -501,4 +535,31 @@ public ViewHolder(View view) {
 
         }
     }
+
+    /**
+     * comapres the filtered string with the exsiting array and creates a new one.
+     *
+     * @param searchtxt
```
Comment (Swati4star) [inline app/src/main/java/io/github/project_travel_mate/destinations/description/PlacesOnMapActivity.java]: add info about `searchtxt`

File: app/src/main/res/layout/bottomsheet_map_cards.xml [hunk 7]
```diff
@@ -13,13 +14,24 @@
...
+    <EditText
+        android:id="@+id/editTextSearch"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_marginStart="8dp"
+        android:layout_marginLeft="12dp"
+        android:layout_marginTop="16dp"
+        android:layout_marginEnd="8dp"
+        android:layout_marginRight="12dp"
+        android:hint="Enter search term" />
```
Comment (Swati4star) [inline app/src/main/res/layout/bottomsheet_map_cards.xml]: dont add constant string in layout file
It should be in strings.xml file

**Comment by smtrz** [comment]
> Also, please share a screenshot with the changes
comment_function


and also i have removed app gradle file from the recent commit.

please review.

**Comment by prabhakar267** [comment]
@smtrz please share the screenshot of the app for us to see the changes without having to build the code locally.

**Comment by smtrz** [comment]
> @smtrz please share the screenshot of the app for us to see the changes without having to build the code locally.

Sure,

here it is :

Screenshot_1571743269


** please use the latest commit.

File: app/src/main/java/io/github/project_travel_mate/destinations/description/PlacesOnMapActivity.java [hunk 8]
```diff
@@ -206,32 +214,18 @@ private void moveMakerToCenter(Marker marker, Double latitude, Double longitude)
     }
 
     /**
-     * on marker selected
-     *
-     * @param marker  marker
+     * @param m = marker
      */
-    private void onPlaceSelected(Marker marker) {
+    private void onPlaceSelected(Marker m) {
```
Comment (Swati4star) [inline app/src/main/java/io/github/project_travel_mate/destinations/description/PlacesOnMapActivity.java]: dont change name of this variable

File: app/src/main/java/io/github/project_travel_mate/destinations/description/PlacesOnMapActivity.java [hunk 9]
```diff
@@ -206,32 +214,18 @@ private void moveMakerToCenter(Marker marker, Double latitude, Double longitude)
     }
 
     /**
-     * on marker selected
-     *
-     * @param marker  marker
+     * @param m = marker
```
Comment (Swati4star) [inline app/src/main/java/io/github/project_travel_mate/destinations/description/PlacesOnMapActivity.java]: revert this back

File: app/src/main/java/io/github/project_travel_mate/destinations/description/PlacesOnMapActivity.java [hunk 10]
```diff
@@ -309,24 +303,39 @@ private void setupViewsAsNeeded(JSONArray feedItems) {
...
     /**
      * Highlights the marker whose card is clicked
-     * @param position position of the marker in
-     *                 mMarkerList whose card is clicked
+     *
+     * @param title this is the title of the marker
      */
-    private void highlightMarker(int position) {
+    private void highlightMarker(String title) {
+        //  Toast.makeText(this, "this is "+title, Toast.LENGTH_SHORT).show();
```
Comment (Swati4star) [inline app/src/main/java/io/github/project_travel_mate/destinations/description/PlacesOnMapActivity.java]: remove

File: app/src/main/java/io/github/project_travel_mate/destinations/description/PlacesOnMapActivity.java [hunk 11]
```diff
@@ -337,29 +346,24 @@ private void highlightMarker(int position) {
...
+        zoomToMarker(currentMarker.getPosition().getLatitude(), currentMarker.getPosition().getLongitude());
     }
 
     /**
      * Zooms in towards the marker whose card is clicked
-     * @param position position of the item in
-     *                 mFeedItems whose card is clicked
+     *
+     * @param latitude
+     * @param longitude
```
Comment (Swati4star) [inline app/src/main/java/io/github/project_travel_mate/destinations/description/PlacesOnMapActivity.java]: add params description

File: app/src/main/java/io/github/project_travel_mate/destinations/description/PlacesOnMapActivity.java [hunk 12]
```diff
@@ -337,29 +346,24 @@ private void highlightMarker(int position) {
...
-        }
+    private void zoomToMarker(double latitude, double longitude) {
+
 
         mController.setZoom(16.0);
         GeoPoint center = new GeoPoint(latitude, longitude);
         mController.animateTo(center);
     }
 
+
```
Comment (Swati4star) [inline app/src/main/java/io/github/project_travel_mate/destinations/description/PlacesOnMapActivity.java]: remove

File: app/src/main/java/io/github/project_travel_mate/destinations/description/PlacesOnMapActivity.java [hunk 13]
```diff
@@ -501,4 +527,31 @@ public ViewHolder(View view) {
...
+    /**
+     * compares the filtered string with the exsiting array and creates a new one.
+     *
+     * @param searchtxt
+     */
+    private void filter(String searchtxt) {
+
+        final JSONArray filteredFeedItems = new JSONArray();
+
```
Comment (Swati4star) [inline app/src/main/java/io/github/project_travel_mate/destinations/description/PlacesOnMapActivity.java]: remove extra new lines

File: app/src/main/res/layout/bottomsheet_map_cards.xml (hunk 7 above)
Comment (smtrz) [reply app/src/main/res/layout/bottomsheet_map_cards.xml]: done

**Comment by smtrz** [comment]
@Swati4star  can you please review and let me know?

**Comment by Swati4star** [comment]
Firstly, Travis checks are failing. They are a must for any PR to be merged. Please resolve that.

File: app/build.gradle (hunk 1 above)
Comment (Swati4star) [reply app/build.gradle]: Why are you making this comment resolved? It's not resolved.
This file should not be a part of commit.

File: build.gradle [hunk 14]
```diff
@@ -22,13 +22,13 @@ allprojects {
     repositories {
         google()
         jcenter()
-        maven { url "https://jitpack.io" }
+
     }
 }
 
 task clean(type: Delete) {
```
Comment (Swati4star) [inline build.gradle]: Why have you moved this file? And made changes to this.

Comment (smtrz) [reply build.gradle]: please allow me sometime to check.

**Comment by smtrz** [comment]
> Firstly, Travis checks are failing. They are a must for any PR to be merged. Please resolve that.

I am unable to find out why travis check is failing, can you please help me out ? would be highly thankful.


this is the log :

* What went wrong:
Task 'checkstyle' not found in root project 'Android'.

**Comment by Swati4star** [comment]
@smtrz You have moved gradle files.
Because of this, the travis is failing.

**Comment by smtrz** [comment]
> @smtrz You have moved gradle files.
> Because of this, the travis is failing.

i was unable to clone so therefore i downloaded the code, i think this might have caused a problem, let me clone it on another machine and send u a pull request again.. thanks for the valuable suggestion.

**Pull request closed without merging**